### DISCUSSION TITLE
ESP32 and Relay switches on cold boot

### DISCUSSION THREAD

**Discussion by Shadesh** [arendst/Tasmota#14310] [Q&A] [answered]
Hi,

i have a question about that combination. I run a ESP32 dev board with one WS2812 and one Relay_i. As soon as I configure the WS2812, the relay switches on for a very short period after performing a cold boot e.g. power loss. I already tried different things, interesting is, if i switch the WS2812 to e.g. 2 and the let the relay be 1 the problem is gone but the WS2812 is not longer available.

Already tried with a fresh install and no config at all, just relay and WS2812, same problem. 

<img width="344" alt="tasmota" src="https://user-images.githubusercontent.com/8633307/148398978-c4554df3-7676-44a3-a7fd-155bf64609e5.png">

What i'm doing wrong here? Thanks a lot!

**Comment by barbudor** [comment, marked as answer]
If you don't configure the  WS2812 at all, this doesn't happen ?

**Reply by Shadesh** [reply]
Hm, its very strange. I cloned the latest dev branch and tried again, now it switches on and off every cold boot without any other configured gpis. Only a relay_i configured.

Yep, installed the tasmota32-bluetooth released via web-installer, same problem. relay switches on and off every cold boot. looks like its not WS2812 related sorry.

**Reply by barbudor** [reply]
It might be hardware.
While the ESP is rebooting , the GPIO are  free floating until Tasmota configure them as an Output
If you have a pull-up resistor on the board, then the signal will go High and Relay_i will switch OFF (a Relay non inverted would switch On)

**Reply by Shadesh** [reply]
I "solved" it for now by using a HIGH trigger relay... so it's not possible with ESP32 and Tasmota because of the hardware design?

**Reply by Jason2866** [reply]
It is not Tasmota, the hardware design is the responsible for.

**Reply by Shadesh** [reply]
Yea looks like. Thanks so far! Will see if I can solved it permanent by using a high trigger relay.

**Comment by Jason2866** [comment]
Please try the standard Tasmota32 version.

**Reply by Shadesh** [reply]
Same problem.

**Reply by Jason2866** [reply]
Tried with a pull up / pull down resistor?
